test/controllers/commit-view-controller.test.js

```diff
@@ -128,5 +147,126 @@ describe('CommitViewController', function() {
       assert.equal(controller.regularCommitMessage, 'regular');
       assert.equal(controller.amendingCommitMessage, 'amending');
     });
+
+    describe('toggling between commit box and commit editor', function() {
+      it('transfers the commit message contents of the last editor', async function() {
+        controller.refs.commitView.editor.setText('message in box');
+
+        commandRegistry.dispatch(atomEnvironment.views.getView(workspace), 'github:toggle-expanded-commit-message-editor');
+        await assert.async.equal(workspace.getActiveTextEditor().getPath(), controller.getCommitMessagePath());
+        await assert.async.isTrue(controller.refs.commitView.props.deactivateCommitBox);
+        const editor = workspace.getActiveTextEditor();
+        assert.equal(editor.getText(), 'message in box');
+
+        editor.setText('message in editor');
+        editor.save();
+
+        commandRegistry.dispatch(atomEnvironment.views.getView(editor), 'pane:split-right-and-copy-active-item');
+        await assert.async.notEqual(workspace.getActiveTextEditor(), editor);
+
+        sinon.spy(controller.refs.commitView, 'update');
+        editor.destroy();
+        await assert.async.isTrue(controller.refs.commitView.update.called);
```

Comment:
🤔 I wonder if this line's intention would be more clear as `await until(() => controller.refs.commitView.update.called)`